src/teleop_twist_joy.cpp

```diff
@@ -74,30 +74,14 @@ TeleopTwistJoy::TeleopTwistJoy(ros::NodeHandle* nh, ros::NodeHandle* nh_param)
   nh_param->param<int>("enable_button", pimpl_->enable_button, 0);
   nh_param->param<int>("enable_turbo_button", pimpl_->enable_turbo_button, -1);
 
-  if (nh_param->getParam("axis_linear", pimpl_->axis_linear_map))
```

Comment:
@dave992 this checks if the param is a map .

Reply:
It however does not ensure that all parameters of the map are set (in this case `scale_linear` and `scale_linear_turbo`), or alternatively provide any default value in case they are missing. 

In [teleop_ps4.yaml](https://github.com/ridgeback/ridgeback/blob/kinetic-devel/ridgeback_control/config/teleop_ps4.yaml) it only sets the yaw axis for example. It then does check for both `scale_linear` and `scale_linear_turbo`, but as they do not exist they use the value resulting from the initialization of the `pimpl_` object. This does not correspond to the default values that are used when the `axis_linear` is not set.